features/FxA-83-sign-in-confirmation-email/README.md

```diff
@@ -0,0 +1,294 @@
+  the solution should be quick to implement.
+
+* The solution must work across all clients,
+  without patches landing in client code.
+
+* The solution must not affect existing sessions
+  from the user's point of view.
+
+* The solution must work from
+  regular Sync sign-in,
+  `/force_auth` and
+  when signing-in to Sync from `/signup`.
+
+## Proposed solution
+
+Add a notion of verification
+to sessionTokens and keyFetchTokens.
+Tokens that are unverified
+have reduced powers (see below).
+Token verification is achieved
+by following a link sent by email.
+
+### What happens when a Sync user signs in?
+
+1. User submits form.
+
+2. Content server requests `POST /account/login?keys=true`.
+
+3. Auth server generates `tokenVerificationId`
+   and sends it with `tokenVerified:false`
+   in request to `PUT /sessionToken/:tokenId`.
+
+4. Auth server sends same `tokenVerificationId`
+   and `tokenVerified:false`
+   in request to `PUT /keyFetchToken/:tokenId`.
+
+5. Auth server invokes the mailer
+   to send a confirmation email,
+   which includes `tokenVerificationId` in the URL.
+
+6. Auth server sends response back to content server,
+   including in the data:
+   ```
+   {
+     "sessionToken": sessionTokenId,
+     "keyFetchToken": keyFetchTokenId,
+     "tokenVerified": false,
+     "challengeReason": "signin",
+     "challengeMethod": "email"
+   }
+   ```
+
+7. Content server navigates to confirm-signin view.
+
+8. Content server starts polling
+   on `GET /token/:tokenId/status`.
+
+9. User clicks link in confirmation email.
+
+10. Content server requests `POST /token/verify`.
+
+11. Auth server requests `POST /token/:tokenVerificationId/verify`.
+    Both tokens are verified in the database.
+
+12. Concurrently:
+
+    * Auth server responds to verification request,
+      including verified tokens in the data.
+
+    * Auth server changes polling request responses,
+      including verified tokens in the data.
+
+13. Content server navigates to settings view.
+
+### What happens when an OAuth user signs in?
+
+1. User submits form.
+
+2. Content server requests `POST /account/login`.
+
+3. Auth server generates `tokenVerificationId`
+   and sends it with `tokenVerified:false`
+   in request to `PUT /sessionToken/:tokenId`.
+
+4. Auth server sends same `tokenVerificationId`
+   and `tokenVerified:false`
+   in request to `PUT /keyFetchToken/:tokenId`.
+
+5. Auth server sends response back to content server,
+   including in the data:
+   ```
+   {
+     "sessionToken": sessionTokenId,
+     "keyFetchToken": keyFetchTokenId,
+     "tokenVerified": false
+   }
+   ```
+
+6. Content server navigates to settings view.
+
+### What happens when an unverified client requests `/certificate/sign`?
+
+If the sessionToken is unverified,
+the request will succeed.
+However, the verification status is stored
+in the certificate as `fxa-tokenVerified`,
+alongside other `fxa-` properties.
+The Sync server can subsequently reject assertions
```

Comment:
This implies a new call to `/certificate/sign` is made once token verification completes. Who makes that call, the browser, or the content server?